tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/AssembleRepositoryMojo.java

```diff
@@ -217,6 +219,26 @@ public class AssembleRepositoryMojo extends AbstractRepositoryMojo {
     @Parameter()
     private boolean addIUTargetRepositoryReferences;
 
+    /**
+     * A list of patterns to exclude repository locations from being added as referenced repository.
+     * <p>
+     * 
+     * <pre>
+     * &lt;excludedRepositoryReferences&gt;
+     *   &lt;excluded&gt;https://foo.bar.org/component1/**&lt;/excluded&gt;
+     *   &lt;excluded&gt;%regex[http(s)?:\/\/bar\.org\/hidden.*]&lt;/excluded&gt;
+     * &lt;/excludedRepositoryReferences&gt;
+     * </pre>
+     * 
+     * The first pattern uses an {@code ANT} pattern and the second pattern uses a
+     * {@code Java RegEx} {@link Pattern} (enclosed in {@code %regex[<the-regex-pattern>]} to
+     * prevent matching locations references from being added as references to the assembled
+     * repository.
+     * </p>
+     */
+    @Parameter
+    private List<String> excludedRepositoryReferences = List.of();
```

Comment:
we should name the parameter something like `<filterReferences>` (just in case someone later wants to add support for `<includes>` or other fancy stuff ...)

Reply:
> we should name the parameter something like `<filterReferences>` (just in case someone later wants to add support for `<includes>` or other fancy stuff ...)

Can you please explain in more detail about what potential methods you are thinking?
Do you mean that one can add positive and negative filters? 

For example maven-surefire also has [excludes](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#excludes) and [includes](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#includes) as dedicated parameters.
If we later support inclusion filters adding a `includedRepositoryReferences` paramter is probably simpler to use than having a `filterReferences` parameter with complex objects that have a extra inclusion/exclusion sub-parameter.

Reply:
yes we should avoid adding more and more dedicated parameters if the belong together.

Reply:
OK. Did that now. And since you mentioned it I also added support for inclusions.
It is now a bit more verbose, but I don't know if/how it could be done more compact:
```
<repositoryReferenceFilters>
	<filter>
		<type>exclusion</type>
		<locationPattern>https://foo.bar.org/hidden/**</locationPattern>
		<patternType>ant</patternType><!-- If not specified, defaults to 'ant'-->
	</filter>
	<filter>
		<type>inclusion</type>
		<locationPattern>http(s)?:\/\/bar\.org\/.*</locationPattern>
		<patternType>regex</patternType>
	</filter>
</repositoryReferenceFilters>
```

Reply:
I don't think it has to be that complicated:

1. we don't need different pattern types, (if at all) a simple prefix matching should be enough
2. `repositoryReferenceFilters` could simply has one list of excludes and one of includes like this:

```
<repositoryReferenceFilters>
	<exclude>https://foo.bar.org/hidden/**</exclude>
	<include>https://foo.bar.org/hidden/butthis</include>
</repositoryReferenceFilters>
```

If you think its to complicated to define as a POJO you can use `PlexusConfiguration` as the parameter type and then get simply the user specified XML there.

Reply:
> I don't think it has to be that complicated:
> 
> 1. we don't need different pattern types, (if at all) a simple prefix matching should be enough

I suggest we simply use the distinction by prefix build into plexus.util.MatchPatterns.
Actually it be more work to forbid it.

> 2. `repositoryReferenceFilters` could simply has one list of excludes and one of includes like this:
> 
> ```
> <repositoryReferenceFilters>
> 	<exclude>https://foo.bar.org/hidden/**</exclude>
> 	<include>https://foo.bar.org/hidden/butthis</include>
> </repositoryReferenceFilters>
> ```

Actually I wanted it that simple but didn't knew/expect that this would work.
I changed it now accordingly and also adapted the example to show the usage for a single and multi element list.